src/touchstone/utils/t_test.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/python
+# also supports python3
+#
+# version 2.0 of this script
+#
+# is_regression.py - statistical test for performance throughput regression
+# based on python scipy.stats.ttest_ind() function
+#
+# we input two sets of samples: 
+#  the baseline sample set -- used as an indication of previously achieved level of performance
+#  the current sample set -- used as an indication of the system currently being tested for performance regression
+#
+# command line inputs:
+#  sample_type -- 'throughput' or 'response-time'
+#  confidence_threshold -- min probability that two sample sets have a different mean 
+#                           (e.g. 95 means that results differ with 95% probability)
+#  max_pct_dev -- maximum percent deviation of either sample set, 100.0 x std.dev/mean 
+#  regression_threshold -- do not return error status unless regression exceeds this percentage
+#  base_sample -- file containing baseline performance throughput samples, 1 per line
+#  current_sample -- file containing current performance throughput samples, 1 per line
+#
+# recommendation: make max_pct_dev 1/2 of regression threshold, so that you can detect small regressions.
+#
+# return status codes and their meanings are described below in the code.
+#  
+# we declare a performance regression if base_set mean is worse than current_set mean and a T-test determines
+# that the probability that the two sample sets have a different mean is greater than confidence_threshold
+#
+# the base sample set mean is "worse" than the current sample set mean if and only if:
+#   the sample_type is 'throughput'    and the base mean > current mean
+#   the sample type is 'response-time' and the base mean < current mean
+#
+# References: The Art of Computer Systems Perf. Analysis, Raj Jain
+# see documentation for python scipy.stats.ttest_ind() function
+#
+# DEPENDENCIES:
+# numpy and scipy RPMs
+# for python3, there are python3-{numpy,scipy} RPMs
+#
+
+import os
+import sys
+from sys import argv, exit
+import math
+import numpy
+import scipy
```

Comment:
we'll need to add these to the https://github.com/cloud-bulldozer/touchstone/blob/master/requirements.txt